hangman.rb

```diff
@@ -0,0 +1,263 @@
+              "| ____|___"
+            ],
+                                                [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |      \\|/ ",
+              "|     |       | ",
+              "|     |      " + "/ ".colorize(:magenta),
+              "|     |",
+              "| ____|___"
+            ],
+                                                [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |      \\|/",
+              "|     |       |",
+              "|     |      / " + "\\ ".colorize(:light_black),
+              "|     |",
+              "| ____|___"
+]]
+
+    start
+  end
+
+  def pick_word
+    word_list = %w{cat dog egg horse donkey elephant pig monkey iguana cow goat unigoat eggplant mouse dragon phoenix}
+    random_word = word_list.sample
+    return random_word
+  end
+
+  def start
+    puts "Welcome to Hangman."
+    puts "Please type your whole word or letter guess. No numbers, punctuation, or spaces."
+    # uncomment the following line for aid in development
+    puts "Answer: #{@game_word}"
+
+    @now_playing = true
+    puts "To exit, type: quit."
+    draw_tree(0)
+   
+    draw_slots
+    draw_wrong_guesses
+
+    play
+  end
+
+  def play
+    while @now_playing
+      guess = get_guess
+      check_guess(guess)
+      update_board
+      check_win
+    end
+  end
+
+  # ---------------------------------
+  # Guess Methods
+
+  def get_guess
+    input = gets.chomp.strip.downcase
+    # allow user to exit game
+    if input == "quit" || input == "exit"
+      exit
+    else
+      guess = validate_input(input)
+    end
+    return guess
+  end
+
+  def validate_input(input)
+    valid = true
+
+    # change valid to false if input string includes anything but a letter
+    input.split("").each do |letter|
+        if !(("a".."z").include?(letter))
+          valid = false
+        end
+      end
+
+    if valid
+      return input
+    else
+      puts "Error:".colorize(:red) + " Please only use letters. No numbers, punctuation, or spaces."
+      get_guess
+    end
+  end
+
+  def check_guess(guess_letter)
+    # check word input
+    if guess_letter.length > 1
+      if guess_letter == @game_word
+        # fill in array of slots with letters of guess word
+        guess_letter.split("").each_with_index do |letter, index|
+          @array_of_slots[index] = letter + " "
+        end
+        # multi-letter guesses that are incorrect count as a mistake
+      else
+        @wrong_letters.push(guess_letter)
+        @errors += 1
+        puts "\nThat is not the correct word."
+      end
+    # guess_letter is just a letter
+    else
+      # check if guess_letter is in game_word
+      if @game_word.include?(guess_letter)
+        # if letter was already figured out, return message
+        if @array_of_slots.join.include?(guess_letter)
```

Comment:
I think this logic around whether or not this letter has already been figured out doesn't need to be nested within the `@game_word` if-statement. If the `@array_of_slots` variable is populated with any letters, it has already passed the `@game_word` check. 
This is not a big deal - but by moving this if statement outside it will help reduce the complexity by removing another potential code path.